test/HdKeyUtils.spec.ts

```diff
@@ -0,0 +1,218 @@
+        },
+        {
+            "path": "44'/1729'/0'/0'/0'",
+            "publicKey": '028551e497c56ce28331172442284c395e218e95e5c00cc243478b7a1119327bf6',
+            "publicKeyHash": 'tz1Suh9rCywdGKfijMc2UdxSHvRvoAczmW2z'
+        },
+        {
+            "path": "44'/1729'/2147483647'",
+            "publicKey": '0263062127bd0b27dbfba16f053701e794cc9e8888d40f90f3e042ce28a7f07bf3',
+            "publicKeyHash": 'tz1SfrAC7bXSpo4LTvngDXcuJHBV3hCFb4bT'
+        },
+        {
+            "path": "44'/1729'/2147483647'/1'",
+            "publicKey": '0287676f2bf5224d9feea3825a565c16521e982626094631a8dcabe8416f50a3a6',
+            "publicKeyHash": 'tz1cTBDJrfYU9t8X7PRagws5o3jT33HsjFBF'
+        },
+        {
+            "path": "44'/1729'/2147483647'/1'/2147483646'",
+            "publicKey": '0201ce46928a4f8555fdcac9f202483077cc926db02cdf05d57555d68eb74a2ada',
+            "publicKeyHash": 'tz1iWtQqrN87VYz9tiQPHu5pDXw4ovM8RJCS'
+        }
+    ]
+}
+
+describe('SLIP10 ed25519 test vectors', () => {
+    it('Trezor test vector 1', async () => {
+        const rootNode = await HDKeyUtils.fromSeed(Buffer.from(ed25519TestVector1.seed, 'hex'), CryptoUtils.ed25519);
+        expect(rootNode.chainCode.toString('hex')).to.equal(ed25519TestVector1.chainCode);
+        expect(rootNode.secretKey.toString('hex')).to.equal(ed25519TestVector1.secretKey);
+        expect(rootNode.publicKey.toString('hex')).to.equal(ed25519TestVector1.publicKey);
+
+        for (const sample of ed25519TestVector1.derivations) {
+            const n = await HDKeyUtils.derivePath(rootNode, sample.path);
+
+            expect(n.chainCode.toString('hex')).to.equal(sample.chainCode);
+            expect(n.secretKey.toString('hex')).to.equal(sample.secretKey);
+            expect(n.publicKey.toString('hex')).to.equal(sample.publicKey);
+        }
+    });
+
+    it('Trezor test vector 2', async () => {
+        const rootNode = await HDKeyUtils.fromSeed(Buffer.from(ed25519TestVector2.seed, 'hex'), CryptoUtils.ed25519);
+        expect(rootNode.chainCode.toString('hex')).to.equal(ed25519TestVector2.chainCode);
+        expect(rootNode.secretKey.toString('hex')).to.equal(ed25519TestVector2.secretKey);
+        expect(rootNode.publicKey.toString('hex')).to.equal(ed25519TestVector2.publicKey);
+
+        for (const sample of ed25519TestVector2.derivations) {
+            const n = await HDKeyUtils.derivePath(rootNode, sample.path);
+
+            expect(n.chainCode.toString('hex')).to.equal(sample.chainCode);
+            expect(n.secretKey.toString('hex')).to.equal(sample.secretKey);
+            expect(n.publicKey.toString('hex')).to.equal(sample.publicKey);
+        }
+    });
+});
+
+describe('Trezor TZ1 address test vector', () => {
+    it('Trezor TZ1 address test vector', async () => {
+        const seed = (await bip39.mnemonicToSeed(trezorTZ1TestVector.mnemonic)).slice(0, 32);
+        const rootNode = await HDKeyUtils.fromSeed(seed, CryptoUtils.ed25519);
+
```

Comment:
If I compute KeyStore like this   
           ```
const n = await HDKeyUtils.derivePath(rootNode, sample.path);

            const kstore = await wrapper.seed_keypair(n.secretKey);

            const keyBuff = Buffer.from(kstore.privateKey.buffer);

            const skey = TezosMessageUtils.readKeyWithHint(keyBuff, 'edsk');
            const store = await KeyStoreUtils.restoreIdentityFromSecretKey(skey);
```
          I'm able to correctly match testVector.